packages/block-editor/src/hooks/align.js

```diff
@@ -248,9 +239,11 @@ addFilter(
 	withDataAlign
 );
 addFilter(
-	'editor.BlockEdit',
+	'editor.BlockControls',
 	'core/editor/align/with-toolbar-controls',
-	withToolbarControls
+	ifCondition( ( { name } ) => hasBlockSupport( name, 'align' ) )(
+		ToolbarControls
+	)
```

Comment:
It's worth noting that these checks aren't contributing much to the performance improvement, but still, it's good practice to avoid as much calls to hooks as possible. Hopefully new hooks and plugins copy this pattern.